### PR TITLE
Make build process more flexible

### PR DESCRIPTION
**Pull request by jakobwenzel**
* Allows overriding Maven repo where to pull rapidwright-api-lib from
* Allows overriding version of rapidwright-api-lib to use
* Split tests into reusable utility classes and actual tests
* Add a testcase that verifies that the open files checker is set up correctly

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -8,11 +8,14 @@ apply from: 'common.gradle'
...
-    line -> line.contains('RAPIDWRIGHT_VERSION:') 
+def read_rapidwright_version() {
+    def sw = new StringWriter()
+    file('.github/workflows/build.yml').filterLine(sw) {
+        line -> line.contains('RAPIDWRIGHT_VERSION:')
+    }
+    return sw.toString().replace("RAPIDWRIGHT_VERSION: v", "").replace("-beta", "").trim() + ""
 }
-def rapidwright_version = sw.toString().replace("RAPIDWRIGHT_VERSION: v", "").replace("-beta", "").trim() + ""
+def rapidwright_version = project.hasProperty('overwrite_rapidwright_version') ? overwrite_rapidwright_version : read_rapidwright_version()
```
Comment (eddieh-xlnx) [inline build.gradle:18]:
```suggestion
def rapidwright_version = project.hasProperty('override_rapidwright_version') ? override_rapidwright_version : read_rapidwright_version()
```

File: build.gradle [hunk 2]
```diff
@@ -174,3 +184,9 @@ test {
   dependsOn testPython
 }
 
+task printVersion() {
```
Comment (eddieh-xlnx) [inline build.gradle:187]:
```suggestion
task printRapidWrightVersion() {
```

File: build.gradle [hunk 3]
```diff
@@ -27,7 +30,12 @@ sourceSets {
...
-      srcDirs = ['test']
+      srcDirs = ['test/src']
+    }
+  }
+  testFixtures {
+    java {
+      srcDirs = ['test/shared']
     }
     resources {
       srcDirs = ['test/testResources']
```
Comment (eddieh-xlnx) [inline build.gradle:41]: A little out of scope and very nitpicky, but can this `test/resources` instead?

Comment (clavin-xlnx) [reply build.gradle:41]: This makes sense to me.  @jakobwenzel do you mind if we refactor this?

Comment (jakobwenzel) [reply build.gradle:41]: Go ahead, I didn't think to rename it after moving it from the root directory.

Comment (clavin-xlnx) [reply build.gradle]: Just refactored it, thanks.

**Review by eddieh-xlnx — APPROVED** [review]

**Pull request merged**